AppCenterDataStorage/AppCenterDataStorageTests/MSDataOperationProxyTests.m

```diff
@@ -0,0 +1,553 @@
+        handler(tokensResponse, nil);
+      }
+      remoteDocumentBlock:^(MSDocumentWrapperCompletionHandler _Nonnull handler) {
+        handler(remoteDocumentWrapper);
+      }
+      completionHandler:^(MSDocumentWrapper *_Nonnull document) {
+        wrapper = document;
+        [expectation fulfill];
+      }];
+
+  // Then
+  [self waitForExpectationsWithTimeout:1
+                               handler:^(NSError *error) {
+                                 if (error) {
+                                   XCTFail(@"Expectation Failed with error: %@", error);
+                                 }
+                                 XCTAssertEqual(wrapper, remoteDocumentWrapper);
+                                 OCMVerify([self.documentStoreMock upsertWithToken:token
+                                                                   documentWrapper:remoteDocumentWrapper
+                                                                         operation:kMSPendingOperationRead
+                                                                  deviceTimeToLive:kMSDataStoreTimeToLiveDefault]);
+                               }];
+}
+
+- (void)testRemoteOperationWhenNoDocumentInStoreAndNoCache {
+
+  // If
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Completed with remote document (no cache)."];
+  __block MSDocumentWrapper *remoteDocumentWrapper = [MSDocumentWrapper alloc];
+  __block MSDocumentWrapper *wrapper;
+  OCMStub([self.documentStoreMock readWithToken:OCMOCK_ANY documentId:OCMOCK_ANY documentType:OCMOCK_ANY])
+      .andReturn([[MSDocumentWrapper alloc] initWithError:self.dummyError documentId:@"documentId"]);
+  MSTokenResult *token = [MSTokenResult alloc];
+  __block MSTokensResponse *tokensResponse = [[MSTokensResponse alloc] initWithTokens:@[ token ]];
+
+  // When
+  [self.sut performOperation:kMSPendingOperationRead
+      documentId:@"documentId"
+      documentType:[NSString class]
+      document:nil
+      baseOptions:[[MSBaseOptions alloc] initWithDeviceTimeToLive:kMSDataStoreTimeToLiveNoCache]
+      cachedTokenBlock:^(MSCachedTokenCompletionHandler _Nonnull handler) {
+        handler(tokensResponse, nil);
+      }
+      remoteDocumentBlock:^(MSDocumentWrapperCompletionHandler _Nonnull handler) {
+        handler(remoteDocumentWrapper);
+      }
+      completionHandler:^(MSDocumentWrapper *_Nonnull document) {
+        wrapper = document;
+        [expectation fulfill];
+      }];
+
+  // Then
+  [self waitForExpectationsWithTimeout:1
+                               handler:^(NSError *error) {
+                                 if (error) {
+                                   XCTFail(@"Expectation Failed with error: %@", error);
+                                 }
+                                 XCTAssertEqual(wrapper, remoteDocumentWrapper);
+                                 OCMReject([[self.documentStoreMock ignoringNonObjectArgs] upsertWithToken:OCMOCK_ANY
```

Comment:
`OCMReject` should be set up in `if` section. Or at least before a call to the rejected method is not supposed to happen. Otherwise the rejection won't happen. Also check other `OCMReject` in this class it looks like other `OCMReject` are misplaced.